### PR TITLE
Build: Bump Iceberg version to 1.5.2

### PR DESCRIPTION
**Pull request by ajantha-bhat**

### CODE REVIEW

File: api-test/src/main/java/org/projectnessie/tools/catalog/migration/api/test/HiveMetaStoreRunner.java
Comment (ajantha-bhat) [inline api-test/src/main/java/org/projectnessie/tools/catalog/migration/api/test/HiveMetaStoreRunner.java:21]: HiveMetastoreTest  is removed due to  https://github.com/apache/iceberg/pull/9282

File: gradle/libs.versions.toml
Comment (ajantha-bhat) [inline gradle/libs.versions.toml:18]: Using 0.79.0 instead of 0.77.1 because it has removed deprecated view classes (so the jar is lighter) and still compatible with Iceberg 1.5.0

File: gradle/libs.versions.toml
Comment (XN137) [inline gradle/libs.versions.toml:3]: "in mapping" means kept in sync?

File: gradle/libs.versions.toml
Comment (XN137) [inline gradle/libs.versions.toml:10]: did these values not change in iceberg 1.5.0 ?

File: gradle/libs.versions.toml
Comment (XN137) [reply gradle/libs.versions.toml:18]: sgtm

generally wondering about compatibility though:

does the catalog-migrator now only support targeting iceberg 1.5.0 ?
if yes do we need to create/update some docs?
if no, do we have backward compatibility tests?

**Review by XN137 — APPROVED** [review]
change lgtm but wondering about compatibility guarantees

File: .github/workflows/main.yml
Comment (XN137) [inline .github/workflows/main.yml:15]: should switch 18 to 21 now maybe?

dropping 11 seems scary imo we should prefer to run an older nessie quarkus server (or make sure even the java11 tests run the server with a newer jdk)

but its your call ultimately

Comment (ajantha-bhat) [reply .github/workflows/main.yml:15]: Removed because of this error. I saw that Nessie PR builders also work with JDK 17 instead of JDK 11.
Could be something to do with Quarkus. 

```
Execution failed for task ':iceberg-catalog-migrator-api:intTest'.
[325](https://github.com/projectnessie/iceberg-catalog-migrator/actions/runs/8434836997/job/23098937620?pr=148#step:4:327)
> Nessie-Server/Quarkus did not emit listen URL. Process: [/usr/lib/jvm/temurin-11-jdk-amd64/bin/java, -Dquarkus.http.port=0, -Dquarkus.management.port=0, -Dquarkus.log.level=INFO, -Dquarkus.log.console.level=INFO, -jar, /home/runner/.gradle/caches/modules-2/files-2.1/org.projectnessie.nessie/nessie-quarkus/0.79.0/6275b00377182632d42386ad95ebcef148ef1d11/nessie-quarkus-0.79.0-runner.jar]
```

Comment (ajantha-bhat) [reply .github/workflows/main.yml:15]: tracking via 
https://github.com/projectnessie/iceberg-catalog-migrator/issues/149

**Review by XN137 — DISMISSED** [review]

File: .github/workflows/main.yml
Comment (snazy) [reply .github/workflows/main.yml:15]: Definitely against dropping Java 11 here. There are definitely ways to make that work. Just removing Java 11 from CI is hacky IMHO.

**Review by snazy — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml
Comment (ajantha-bhat) [reply .github/workflows/main.yml]: ok. Using `0.74.0` if the java version is older than 17

**Review by XN137 — APPROVED** [review]
lgtm

File: .github/workflows/main.yml
Comment (ajantha-bhat) [reply .github/workflows/main.yml]: @snazy: Please let me know if the updated changes are ok.

Comment (snazy) [reply .github/workflows/main.yml]: No, not okay. Downgrading is also not an option IMO.

Why don't you run the server w/ Java 17? The plugin supports this, no?

Comment (snazy) [reply .github/workflows/main.yml]: > I don't know How the PR builder can automatically install JDK 17.

We are doing this in other builds - maybe look in the docs for the GH action?

**Comment by snazy** [comment]
Any progress?

**Comment by ajantha-bhat** [comment]
> Any progress?

@snazy: Sorry, I have lost email notification on this one. Updated the PR now.

File: .github/workflows/main.yml
Comment (ajantha-bhat) [reply .github/workflows/main.yml]: Which project? I checked Nessie, it was using only JDK 17 and above and couldn't find anything related on query-engine-integration-tests either. 

I configured JDK 17 for all java version related tests. But couldn't specify the nessieQuarkusServer to use JDK 17 always.

**Comment by snazy** [comment]
@ajantha-bhat could you please verify that your changes work locally.

> But couldn't specify the nessieQuarkusServer to use JDK 17 always.

There _is_ a way to tell the Java-version - please take at least a look into the options for the Gradle plugin.

**Comment by ajantha-bhat** [comment]
@XN137, @snazy: PR is ready (finally)

**Comment by ajantha-bhat** [comment]
Once this PR is merged, I can do a new release. I merged all other open PRs.

**Review by snazy — APPROVED** [review]

**Pull request merged**
